bridge_adaptivity/module/mixins.py

```diff
@@ -30,7 +31,12 @@ def dispatch(self, request, *args, **kwargs):
         sequence_id = request.session.get('Lti_sequence')
         if not lti_session:
             log.error('Lti session is not found, Request cannot be processed')
-            return HttpResponseForbidden("Cource content is available only through LTI protocol.")
+            msg = """
```

Comment:
To ensure the unit is opened through the LTI initial string must be left and extended with your text. I would only exchange the URL to BRIDGE_HOST. And add possibility with cookies security decreasing to allow all for start working on the adaptive unit and increase it back after unit starts to render correctly.

@kunanit Andrew could you please explain why do we need to visit `https://courses.openedx.vpal.io/event` before new bridge session creation?

Reply:
It appeared that once the learner has a session by visiting the bridge in a new tab, are able to access the sequence view, and they try to submit a new problem, clicking the "Submit" button (this is in the content source iframe) appears to not do anything. On click, it triggers these errors in the browser console:

<img width="1006" alt="screen shot 2017-10-12 at 11 04 57 am" src="https://user-images.githubusercontent.com/7773522/31503469-bd3a1158-af3d-11e7-94be-3775d2ee8f17.png">

I figured it was the same kind of problem, so visiting the content source in a new tab and reloading the LTI tool solved this. I chose `/event` on the content source so that learners didn't get confused and try to sign up for accounts on the content source openedx instance.

Reply:
Oh, got it, thanks. I didn't know that OpenEdx uses the session to check submitted answers.

Reply:
it should be better to raise exception (in case if it should return HTML response, but with status code for example 400 or 403)

```
from django.core.exceptions import  SuspiciousOperation, PermissionDenied, Http404
from django.http.response import Http404

# to return HttpResponse with status code 400
raise SuspiciousOperation("Bad request")
# you may open django.core.exceptions file and find exceptions inherited from 
# SuspiciousOperation class, they are also OK to convert to response 400.

# to return HttpResponse with status code 403
raise PermissionDenied('You can not view this content')

# to return HttpResponse with status code 404
raise Http404('Page not found')

# to return HttpResponse with status code 500
raise Exception()
```

django will catch exception and handle it properly, so it will return correct response rendered into correct template.

[How django handles errors](https://docs.djangoproject.com/en/2.0/topics/http/urls/#error-handling), and [here](https://docs.djangoproject.com/en/2.0/ref/views/#error-views)